docs/contributor/02-10-testing-strategy.md

```diff
@@ -0,0 +1,65 @@
+# Testing strategy of the Istio module
+
+## Background
+The Istio module requires a testing strategy to ensure that the module is always functioning
+as expected on all supported platforms.
+However, running tests on all supported platforms in all cases is both time-consuming and expensive.
+Therefore, we need to decide on a testing strategy that balances the need for comprehensive
+testing with the requirement for fast feedback and development.
+
+## Testing strategy outline
+
+The testing strategy for the Istio module will be implemented according to the following guidelines:
+1. Tests that depend on the Gardener platform will not run on Pull Requests (PRs).
+2. Gardener related tests will run during post-merge workflows and on scheduled runs.
+3. If a Gardener-related test fails, the cluster will remain alive for debugging purposes.
+4. In the event of a test failure during a post-merge workflow, the PR owner is responsible for resolving the issue.
+5. PR tests should generally avoid relying on external resources.
+   This especially means that secrets should not be required for running PR tests whenever possible.
```

Comment:
Let's maybe add one sentence more to explain that having secrets and external resources is a security concern as this is an open-source project.